### PR TITLE
dataframe.filter should preserve order of items

### LINKED ISSUE #245: DataFrame.filter(axis="index", items=[...]) should preserve order of items

### PR DESCRIPTION
**Pull request by V1NAY8**
Closes #245 
- Added typehints wherever necessary w.r.t filter 
Example:
```python
>>> ed_flights = ed.DataFrame("localhost", "flights")
>>> ed_flights.filter(axis="index", items = ["1", "2", "0"])
   AvgTicketPrice  Cancelled           Carrier                                          Dest  ... OriginRegion OriginWeather dayOfWeek           timestamp
1      882.982662      False  Logstash Airways                     Venice Marco Polo Airport  ...        SE-BD         Clear         0 2018-01-01 18:27:00
2      190.636904      False  Logstash Airways                     Venice Marco Polo Airport  ...        IT-34          Rain         0 2018-01-01 17:11:14
0      841.265642      False   Kibana Airlines  Sydney Kingsford Smith International Airport  ...        DE-HE         Sunny         0 2018-01-01 00:00:00
```
@sethmlarson Please review. 😄

### CODE REVIEW

**Review by sethmlarson — CHANGES_REQUESTED** [review]
This mostly looks good to me, thanks! One comment:

**Comment by sethmlarson** [comment]
jenkins test this please

**Comment by V1NAY8** [comment]
There's a recent mypy release which is causing issues
I see from build output `eland/utils.py:70: error: Value of type variable "_LT" of "sorted" cannot be "Item". `
Trying to resolve this.

**Comment by V1NAY8** [comment]
I have resolved the mypy issue.

File: eland/utils.py [hunk 1]
```diff
@@ -62,7 +61,7 @@ def to_list(x: Union[Collection[Any], pd.Series]) -> List[Any]:
     raise NotImplementedError(f"Could not convert {type(x).__name__} into a list")
 
 
-def try_sort(iterable: Iterable[Item]) -> Iterable[Item]:
+def try_sort(iterable: Iterable[str]) -> Iterable[str]:
```
Comment (sethmlarson) [inline eland/utils.py:64]: This works for us I believe but I wonder what the general solution to this issue is.

**Review by sethmlarson — APPROVED** [review]
LGTM, will merge once test suite passes :)

**Comment by sethmlarson** [comment]
jenkins test this please

File: eland/utils.py (hunk 1 above)
Comment (V1NAY8) [reply eland/utils.py:64]: I tried looking for solution, Tried a lot of combinations :) But couldnt find any. 😕

**Comment by V1NAY8** [comment]
Thanks, I need to work on naming variables! 😮

**Comment by sethmlarson** [comment]
@V1NAY8 Naming is one of the famous unsolved problems in software engineering ;)

**Comment by V1NAY8** [comment]
Could you please add hacktoberfest-accepted label ?  Thanks

**Pull request merged**